docs/src/main/asciidoc/http-reference.adoc

```diff
@@ -546,7 +546,135 @@ link:https://undertow.io/undertow-docs/undertow-docs-2.0.0/index.html#predicates
 
 If you are using a `web.xml` file as your configuration file, you can place it in the `src/main/resources/META-INF` directory.
 
-=== Built-in route order values
+[[vertx-web-sessions]]
+== Sessions
+
+Quarkus includes support for sessions, based on https://vertx.io/docs/vertx-web/java/#_handling_sessions[Vert.x Web sessions].
+
+By default, sessions are disabled.
+To enable them, set the `quarkus.http.sessions.mode` configuration property to:
+
+`in-memory`:: for sessions stored in memory of the Quarkus application
+`redis`:: for sessions stored in an external Redis server; requires using the Quarkus Redis Client extension
+`infinispan`:: for sessions stored in an external Infinispan data grid; requires using the Quarkus Infinispan Client extension
+
+This configuration property is fixed at build time and cannot be changed at runtime.
+
+WARNING: Undertow includes its own support for sessions.
+If Undertow is present, Vert.x Web sessions cannot be enabled.
+
+Sessions require using a cookie, which holds the session identifier.
+By default, the cookie name is `JSESSIONID`.
+Cookieless sessions are not supported.
+Storing session data directly into the session cookie is not supported either.
+
+=== Accessing sessions
+
+When sessions are enabled, the Vert.x Web `Session` object may be obtained from the current `RoutingContext` using `RoutingContext.session()`.
+Alternatively, the `io.vertx.ext.web.Session` object may be injected.
+
+When using non-clustered in-memory sessions, arbitrary objects may be stored into a session.
+
+With cluster-wide in-memory sessions, Redis, or Infinispan, the following data types may be stored into a session:
+
+* primitive wrapper types
+** `java.lang.Boolean`
+** `java.lang.Byte`
+** `java.lang.Short`
+** `java.lang.Integer`
+** `java.lang.Long`
+** `java.lang.Float`
+** `java.lang.Double`
+** `java.lang.Character`
+* `java.lang.String`
+* `byte[]`
+* `io.vertx.core.buffer.Buffer`
+* implementations of `io.vertx.core.shareddata.ClusterSerializable`
+
```

Comment:
What about JsonObject and JsonArray? Do you have to transform them into buffer before?

Reply:
`JsonObject` and `JsonArray` are actually `ClusterSerializable`, so they work just fine.

There's a few more classes in Vert.x that also implement `ClusterSerializable`, but those are not terribly interesting outside of Vert.x. I'll replace "implementations of `io.vertx.core.shareddata.ClusterSerializable`" with `JsonObject` and `JsonArray`.

Reply:
Done.